### PR TITLE
Add r-gsdesign package.

### PR DESCRIPTION
**Pull request by matthewwardrop**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by matthewwardrop** [comment]
@keaven Here is the conda-forge package spec as described in https://github.com/keaven/gsDesign/issues/24 . Would you like to be added as a maintainer?

**Comment by matthewwardrop** [comment]
@conda-forge/help-r This should be ready for review. All the other R feedstocks I've seen have been directly looked after by `@conda-forge/r`; so if this is something you'd like to do here, I'd be more than happy for you to replace the maintainer field appropriately.

**Comment by matthewwardrop** [comment]
@conda-forge/help-r How do we enforce the R >=3.5 constraint for arch-specific R packages?

File: recipes/r-gsdesign/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+build:
+  merge_build_host: true  # [win]
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+
+requirements:
+  host:
+    - r-base >=3.5.0
```
Comment (cbrueffer) [inline recipes/r-gsdesign/meta.yaml:24]:
```suggestion
    - r-base
```

File: recipes/r-gsdesign/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+test:
+  commands:
+    - $R -e "library('gsDesign')"           # [not win]
+    - "\"%R%\" -e \"library('gsDesign')\""  # [win]
+
+about:
+  home: https://cran.r-project.org/web/packages/gsDesign/index.html, https://github.com/keaven/gsDesign
+  license: GPL-3.0-or-later
+  license_family: GPL
+  license_file: LICENSE
```
Comment (cbrueffer) [inline recipes/r-gsdesign/meta.yaml:49]:
```suggestion
  license_file:
    - {{ environ["PREFIX"] }}/lib/R/share/licenses/GPL-2
    - LICENSE
```

File: recipes/r-gsdesign/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  summary: Derives group sequential designs and describes their properties.
+
+extra:
+  recipe-maintainers:
+    - matthewwardrop
```
Comment (cbrueffer) [inline recipes/r-gsdesign/meta.yaml:54]:
```suggestion
    - conda-forge/r
    - matthewwardrop
```

File: recipes/r-gsdesign/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  host:
+    - r-base >=3.5.0
+    - r-ggplot2 >=0.9.2
+    - r-xtable
+    - r-plyr
+    - r-dplyr
+    - r-mvtnorm
+    - r-tibble
+  run:
+    - r-base >=3.5.0
```
Comment (cbrueffer) [inline recipes/r-gsdesign/meta.yaml:32]:
```suggestion
    - r-base
```

File: recipes/r-gsdesign/meta.yaml (hunk 1 above)
Comment (cbrueffer) [reply recipes/r-gsdesign/meta.yaml:24]: conda-forge only builds for R 3.6 and later.

File: recipes/r-gsdesign/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  sha256: d2c75dfbb4a87b766a8164c3a8b155169c2df3c1b096b1babc2a8dbb710e559f
+
+build:
+  merge_build_host: true  # [win]
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+
+requirements:
```
Comment (cbrueffer) [inline recipes/r-gsdesign/meta.yaml:22]:
```suggestion
requirements:
  build:
    - {{ compiler('c') }}              # [not win]
    - {{ compiler('m2w64_c') }}        # [win]
    - {{ posix }}filesystem        # [win]
    - {{ posix }}make
    - {{ posix }}sed               # [win]
    - {{ posix }}coreutils         # [win]
    - {{ posix }}zip               # [win]
```

**Comment by matthewwardrop** [comment]
@cbrueffer I've removed myself as a maintainer, since I'm quite happy to cede maintainership :).

**Comment by cbrueffer** [comment]
No problem! Feel free to add yourself any time if you feel like it.

**Pull request merged**